### PR TITLE
Make processor info strings translatable

### PR DESCRIPTION
**Pull request by bp0**
Break up the big sprintf() format string for processor (x86) that was translated as one piece into smaller more re-usable pieces. 
* the recently added topology and cpufreq information labels can be translated
* common strings like "Processor", "Capabilities", "Frequency", etc
* "Little Endian" and "Big Endian"
* some unit labels "kHz", "MHz", "ns" (maybe not so useful ... or maybe so, it seems that Russian MHz is different)

Only tested on x86 and ARM hardware so far. For all others, /proc/cpuinfo reading has been tested, but not on actual hardware.

Update:
* modules/devices/*/processor.c are all cleaned up. All strings are translatable.
* Added CMake options and a test script to test that code without the actual hardware.
* all processor.c are tested and working
* riscv added
* ia64 improved
* parisc repaired and improved

### CODE REVIEW

**Comment by mckaygerhard** [comment]
after that pull i can help with the missing string on spanish and some russain

**Comment by bp0** [comment]
@mckaygerhard, right on.

File: modules/devices/ia64/processor.c [hunk 1]
```diff
@@ -19,36 +19,48 @@
 #include "hardinfo.h"
 #include "devices.h"
 
+gchar *byte_order_str() {
```
Comment (lpereira) [inline modules/devices/ia64/processor.c]: This can be moved to somewhere common.

File: modules/devices/ia64/processor.c [hunk 2]
```diff
@@ -19,36 +19,48 @@
...
+#if G_BYTE_ORDER == G_LITTLE_ENDIAN
+    return _("Little Endian");
+#else
+    return _("Big Endian");
+#endif
+}
+
+#ifndef PROC_CPUINFO
+#define PROC_CPUINFO "/proc/cpuinfo"
```
Comment (lpereira) [inline modules/devices/ia64/processor.c]: Same for this.

**Comment by lpereira** [comment]
Excellent, thanks! Just a few minor nitpicks, though.

**Comment by bp0** [comment]
I'm done for now. This could be merged, if you like.

**Comment by bp0** [comment]
Regarding the flag definitions, it really seems like there needs to be some additional info for the translator.
There are a couple ways that might be done.

Use code comments. 
* Adjust xgettext params, like `xgettext -ctrans ...`
* Add comments in the code:
> ...
>  `{ "fpu",	N_(/* trans: x86 flag fpu*/ "Onboard FPU (floating point support)") },` 
> ...
* Result in the .po:
> #. trans: x86 flag fpu
> #: modules//devices/x86/processor.c:339
> msgid "Onboard FPU (floating point support)"
> msgstr ""`

Alternately, gettext context could be provided, using pgettext(), but this is a little more complicated when it comes to calling gettext. See [gettext contexts](https://www.gnu.org/software/gettext/manual/html_node/Contexts.html)

**Comment by bp0** [comment]
Regarding using gettext contexts and comments for translatable flag definitions,
It looks like this:
* https://github.com/bp0/rpiz/blob/trans/src/arm_data.c#L42
* https://github.com/bp0/rpiz/blob/trans/src/riscv_data.c#L38

A snippet of the .po(t) result:
>#. /ext:vfpv4
>#: arm_data.c:58
>msgctxt "arm-flag"
>msgid "VFP version 4 with fast context switching"
>msgstr ""

context arm-flag, and comment ext:vfpv4 so it is known what the string is supposed to define.

**Comment by bp0** [comment]
The problems with the special fields "..." are resolved. Solves the "unknown field" portion of #38.

**Comment by mckaygerhard** [comment]
this pull request must be renamed to "translatable strings" due solves other issues like  #38 and #31 
please merge and then close that issues! a picture to demostrate (non tranlated its due are missing in po file)
hardinfomemetranslatable

**Comment by Obliteron** [comment]
> I suggest uploading the source code of the language in Transifex, so everyone can contribute to the translation!
> Eu sugiro fazer o upload do código-fonte do idioma no Transifex, assim todo mundo pode contribuir para a tradução!
@lpereira

**Comment by Obliteron** [comment]
>>> I suggest uploading the source code of the language in Transifex, so everyone can contribute to the translation!

>>> Eu sugiro fazer o upload do código-fonte do idioma no Transifex, assim todo mundo pode contribuir para a tradução!
@lpereira

**Comment by mckaygerhard** [comment]
well github its already a colaboration tool, and the po files inside the repository can be handle locally by linux po trasnaltion tools, one the git repo are cloned, so the tranfiex "layer" can complicated simple things...

**Comment by bp0** [comment]
Oh, that last one conflicts with the fix for libc detection. I was trying to avoid that, but I messed up. I'll stop now until this set is merged.

**Comment by lpereira** [comment]
Merged, thank you very much!

**Pull request closed without merging**